holoviews/core/util.py

```diff
@@ -933,8 +933,15 @@ def max_range(ranges, combined=True):
         with warnings.catch_warnings():
             warnings.filterwarnings('ignore', r'All-NaN (slice|axis) encountered')
             values = [tuple(np.NaN if v is None else v for v in r) for r in ranges]
-            if pd and all(isinstance(v, pd.Timestamp) for r in values for v in r):
-                values = [(v1.to_datetime64(), v2.to_datetime64()) for v1, v2 in values]
+            if pd and any(isinstance(v, datetime_types)
+                          for r in values for v in r):
+                values = [
+                    (pd.Timestamp(i).to_datetime64(),
+                     pd.Timestamp(j).to_datetime64())
+                    if isinstance(i, datetime_types)
+                    and isinstance(j, datetime_types)
```

Comment:
I also think this probably needs special handling for `cftime` types which have all kinds of additional complexities.

Reply:
For now it's probably best to make this code ignore cftime types.